parachain/integration-test/src/lib.rs

```diff
@@ -0,0 +1,379 @@
+			runtime_types::pallet_ismp_demo::pallet::{GetRequest, TransferParams},
+		},
+	},
+	Hyperbridge,
+};
+use tesseract::logging::setup as log_setup;
+use tesseract_messaging::relay;
+use tesseract_primitives::{config::RelayerConfig, IsmpProvider};
+use tesseract_substrate::{config::KeccakSubstrateChain, SubstrateClient, SubstrateConfig};
+use transaction_fees::TransactionPayment;
+
+/// This function is used to fetch get request and construct get response and submit to chain
+/// A(source chain)
+async fn messaging_relayer_lite(
+	chain_a_sub_client: SubstrateClient<Hyperbridge>,
+	chain_b_sub_client: SubstrateClient<Hyperbridge>,
+	tx_block_height: u64,
+) -> Result<Vec<u8>, anyhow::Error> {
+	let (client_a, client_b) =
+		(chain_a_sub_client.clone().client, chain_b_sub_client.clone().client);
+
+	let (chain_a_client, chain_b_client) = (
+		Arc::new(chain_a_sub_client) as Arc<dyn IsmpProvider>,
+		Arc::new(chain_b_sub_client) as Arc<dyn IsmpProvider>,
+	);
+
+	let state_machine_update = StateMachineUpdated {
+		state_machine_id: chain_a_client.state_machine_id(),
+		latest_height: tx_block_height,
+	};
+	let event_result_a = chain_a_client
+		.query_ismp_events(tx_block_height - 1, state_machine_update.clone())
+		.await?;
+
+	// ====================== get the GET_REQUEST ===============================
+	let get_request = event_result_a
+		.into_iter()
+		.find_map(|event| match event {
+			Event::GetRequest(_) => Some(event),
+			_ => None,
+		})
+		.expect("Expected at least one GetRequest event");
+	// ======== process the request offchain ( 1. Make the response ) =============
+
+	let response = {
+		let get = match get_request {
+			Event::GetRequest(get) => get,
+			_ => panic!("Not supported"),
+		};
+		let dest_chain_block_hash = client_b.rpc().block_hash(Some(get.height.into())).await?;
+		let keys = get.keys.iter().map(|key| &key[..]).collect::<Vec<&[u8]>>();
+		let value_proof = client_b.rpc().read_proof(keys, dest_chain_block_hash).await?;
+		let proof = value_proof.proof.into_iter().map(|bytes| bytes.0).collect::<Vec<Vec<u8>>>();
+
+		let proof_of_value = SubstrateStateProof::StateProof(StateMachineProof {
+			hasher: HashAlgorithm::Keccak,
+			storage_proof: proof,
+		});
+		let proof = Proof {
+			height: StateMachineHeight {
+				id: chain_b_client.state_machine_id(),
+				height: get.height,
+			},
+			proof: proof_of_value.encode(),
+		};
+		let response = ResponseMessage {
+			datagram: RequestResponse::Request(vec![Get(get.clone())]),
+			proof,
+			signer: chain_a_client.address(), // both A&B have same relayer address
+		};
+
+		Message::Response(response)
+	};
+	// =================== send to the source chain ================================
+	let _res = chain_a_client.submit(vec![response]).await?;
+	//==================== after approx 7-9 blocks the response event is emitted ===
+	// =================== fetch the returned value ================================
+	tokio::time::sleep(Duration::from_secs(30)).await;
```

Comment:
Watch for the event, otherwise you'll miss the event and this test will be flaky